src/mesh/generic/Boundary.cpp

```diff
@@ -83,4 +99,73 @@ double Boundary::interpolate_prescribed_load(std::vector<std::pair<double, doubl
 
     return (1.0 - interpolation_factor) * load_0 + interpolation_factor * load_1;
 }
+
+void Boundary::generate_sinusoidal(Json::Value const& boundary, std::string const& name)
+{
+    using namespace ranges;
+
+    Json::Value const& Amplitude = boundary["Amplitude"][name];
+    Json::Value const& Period = boundary["Period"];
+    Json::Value const& NumberOfCycles = boundary["NumberOfCycles"];
+    Json::Value const& TimeStep = boundary["TimeStep"];
+    Json::Value const& Phase = boundary["Phase"];
+
+    std::vector<double> amplitude = Amplitude | view::transform([](auto const& element) {
+                                        return element.asDouble();
+                                    });
+    std::vector<double> period = Period | view::transform([](auto const& element) {
+                                     return element.asDouble();
+                                 });
+    std::vector<double> number_of_cycles = NumberOfCycles | view::transform([](auto const& element) {
+                                               return element.asDouble();
+                                           });
+    std::vector<double> time_step = TimeStep | view::transform([](auto const& element) {
+                                        return element.asDouble();
+                                    });
+    std::vector<double> phase = Phase | view::transform([](auto const& element) {
+                                    return element.asDouble();
+                                });
+
+    if (!((amplitude.size() == period.size()) && (period.size() == number_of_cycles.size())
+          && (number_of_cycles.size() == time_step.size()) && (time_step.size() == phase.size())))
+    {
+        throw std::runtime_error("The amplitude, period, number_of_cycles, time_step and phase "
+                                 "vectors must be of the same size.");
+    }
+
+    std::vector<double> new_times;
+    std::vector<double> new_loads;
+    new_times.push_back(0);
+
+    // Loop over the blocks
+    for (int i = 0; i < amplitude.size(); ++i)
+    {
+        double omega = 2.0 * M_PI / period[i];
+        auto number_of_steps = number_of_cycles[i] * period[i] / time_step[i];
+
+        std::vector<double> time_block(number_of_steps);
+        std::iota(time_block.begin(), time_block.end(), 1.0);
```

Comment:
```std::iota``` is usually for integer sequences.  We can just use 1 here or a generate function if you care:

```cpp
    std::generate(v.begin(), v.end(), [n = 0.0] () mutable { return n += 1.0; });
```